compiler/tests/fixtures/generator.py

```
@@ -0,0 +1,317 @@
+    try:
+        yield docker
+    finally:
+        await docker.close()
+
+
+async def docker_available() -> bool:
+    """Check if Docker is available and running."""
+    try:
+        async with docker_client() as docker:
+            await docker.version()
+        return True
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)